### PR TITLE
feat: Support typed properties and attributes in `fully_qualified_strict_types`

### LINKED ISSUE #5004: Support typed properties in `fully_qualified_strict_types`
Currently `fully_qualified_strict_types` can handle multiple scenarios, but does not cover class' properties. We need this fixer to make changes like this:

```diff
 use Foo\Bar;
 
 class SomeClass
 {
-    private \Foo\Bar $baz;
+    private Bar $baz;
 }
```

### LINKED ISSUE #7657: `fully_qualified_strict_types` must fix attributes
## Feature request

Expected fixing behaviour:
```diff
-#[\Attr]
+#[Attr]
 class Cl {}
```

### PR DESCRIPTION
**Pull request by mvorisek**
fix #5004
fix #7657

### CODE REVIEW

**Comment by Wirone** [comment]
Side note: I believe `shortenClassIfPossible` and `replaceClassWithShort` can be merged somehow, the fact we have 2 methods for that comes from #7459 which was a PR revived from an older branch, and for me it was more important to support new cases than solve this puzzle 😉. I thought we could refactor it later seamlessly, so @mvorisek if you currently work so much on this fixer, maybe you can verify if my idea is valid 🙂. To be clear: it has nothing to do with this PR.

**Comment by mvorisek** [comment]
You won't be surprised I have some related code stashed on my PR... It should be solved together with #7619. Yes, it is unrelated.

**Comment by mvorisek** [comment]
@Wirone can this PR be merged? I need to to fix/finish #7679 as all short types must be implicitly imported.

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php [hunk 1]
```diff
@@ -353,6 +353,31 @@ class SomeClass extends \Foo\Bar\A implements \Foo\Bar\Izumi, A, \A\B, \Foo\Bar\
             '<?php use A\B; class Foo { use \A\C; use \D; use \A\B { \A\B::bar as baz; } };',
         ];
 
+        yield 'typed property in function' => [
+            '<?php use A\B; function foo(B $v) {}',
+            '<?php use A\B; function foo(\A\B $v) {}',
+        ];
```
Comment (Wirone) [inline tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:359]: It was fixed already and is covered with tests, no need to add another case.

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php [hunk 2]
```diff
@@ -353,6 +353,31 @@ class SomeClass extends \Foo\Bar\A implements \Foo\Bar\Izumi, A, \A\B, \Foo\Bar\
...
+
+        yield 'typed property in class' => [
+            '<?php use A\B; class Cl { public B $p; }',
+            '<?php use A\B; class Cl { public \A\B $p; }',
+        ];
+
+        yield 'typed promoted property in class' => [
+            '<?php use A\B; class Cl { public function __construct(B $p) {} }',
+            '<?php use A\B; class Cl { public function __construct(\A\B $p) {} }',
```
Comment (Wirone) [inline tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:368]:
```suggestion
            '<?php use A\B; class Cl { public function __construct(private B $p) {} }',
            '<?php use A\B; class Cl { public function __construct(private \A\B $p) {} }',
```

File: tests/Fixtures/Integration/misc/PHP8_0.test [hunk 3]
```diff
@@ -30,7 +32,7 @@ $country = $session?->user?->getAddress()?->country;
 // https://wiki.php.net/rfc/shorter_attribute_syntax
 // https://wiki.php.net/rfc/shorter_attribute_syntax_change
 #[MyAttribute]
-#[\MyExample\MyAttribute]
+#[MyAttribute2]
```
Comment (Wirone) [inline tests/Fixtures/Integration/misc/PHP8_0.test:35]: I believe that `#[MyAttribute]` and `#[\MyExample\MyAttribute]` was here to show that name conflict prevents the latter from being imported, so please keep it as it was before. If you need to import another attibute's FQCN, please add new attribute.

**Review by Wirone — CHANGES_REQUESTED** [review]

**Comment by Wirone** [comment]
@mvorisek PR was marked as draft so I did not review it 🤷‍♂️ 😉.

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php (hunk 1 above)
Comment (mvorisek) [reply tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:359]: what test case name do you mean to remove and where it is already tested?

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php (hunk 2 above)
Comment (mvorisek) [reply tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php]: 👍 to test promoted property really, done

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php (hunk 1 above)
Comment (mvorisek) [reply tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php]: removed

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you veeeery much @mvorisek, great work 🍻!

**Pull request merged**